Migrate to Cmake and ensure `-fstack-protector-all` is being applied

scottyab opened the pull request:
Resolves #170 

* Migrate to Cmake and use the correct `-fstack-protector-all` flag, Thanks @stealthcopter for investigating the correct flag and verifying with checksec
* Remove the compiled binaries and gradle config that used to allow optional NDK - this is now required if anyone is building the library as I felt these potentially risk old binaries being used and not fresh ones

stealthcopter reviewed (approved):
LGTM

nfmobile commented:
Dear  @scottyab  and @stealthcopter  ,

Thank you for your support, we have noticed that the .so files has been removed  to fix the security issue that we have opened : https://github.com/scottyab/rootbeer/issues/170 

Please could you inform us by when this fix will be released ? we have tried today to reinstall the plugin from current master 0.0.9 and the .so files are still present and imported into our APK under lib folder.

Awaiting your feedback
Thank you

scottyab commented:
Hi @nfmobile 

>we have noticed that the .so files has been removed to fix the security issue that we have opened

I think they might be some misunderstanding. You are correct we have removed the prebuild .so files from the src/ directory but not to address the issue. To address the -fstack flag issue we moved to CMake and added the correct compiler flag.

For some historical content we added the pre-build .so files into the GitHub repo a while back to allow people without the NDK to build the library but I was concerned this could lead to us accidentally releasing builds with out of date *.so files. So now when we build a release we will have to generate fresh versions and require the NDK (note if you're using the aar file nothing has changed). 

So the fix for the correct -fstack flag library will be in version 0.1.0 which I'll be uploading to maven central this morning. Hope this helps clarify things, but if not please do let us know

nfmobile commented:
@scottyab Thank you for your clarification. We will be waiting for version 0.1.0. Thanks again for your swift support.

scottyab commented:
I've uploaded/released 0.1.0 to maven central, it should be available in the next few hours.

nfmobile commented:
@scottyab  it's working now and stack canary are now available in so files. the warning is not appearing any more.
thanks so much 👍

scottyab commented:
Also kudos @stealthcopter too for finding -fstack-protector-all worked where as -fstack-protector-strong didn't!
